valhalla/baldr/rapidjson_utils.h

```diff
@@ -148,6 +153,74 @@ inline boost::optional<rapidjson::Value&> get_child_optional(V&& v, const char*
   return c;
 }
 
+template <class Ptree> void add_value(const Value& v, Ptree& pt) {
+  switch (v.GetType()) {
+    case kObjectType:
+      add_object(v.GetObject(), pt);
+      break;
+    case kArrayType:
+      add_array(v.GetArray(), pt);
+      break;
+    case kNullType:
```

Comment:
any case from here down is a base case that stops the recursion. the calls to add_object and add_array deepen the call stack